File: java/program/Barang.java
import java.util.Scanner;

public class Barang {
   // ini atribut barang, isinya data dasar yang biasanya dimiliki barang
   private String kode;
   private String nama;
   private String harga;
   private String merk;
   private String garansi;
   private String stok;

   // constructor, jadi pas bikin objek langsung bisa masukin data
   public Barang(String kode, String nama, String harga, String merk, String garansi, String stok) {
      this.kode = kode;
      this.nama = nama;
      this.harga = harga;
      this.merk = merk;
      this.garansi = garansi;
      this.stok = stok;
   }

   // di bawah ini getter sama setter, biar bisa ambil atau ubah data
   public String getKode() {
      return this.kode;
   }

   public void setKode(String kode) {
      this.kode = kode;
   }

   public String getNama() {
      return this.nama;
   }

   public void setNama(String nama) {
      this.nama = nama;
   }

   public String getHarga() {
      return this.harga;
   }

   public void setHarga(String harga) {
      this.harga = harga;
   }

   public String getMerk() {
      return this.merk;
   }

   public void setMerk(String merk) {
      this.merk = merk;
   }

   public String getGaransi() {
      return this.garansi;
   }

   public void setGaransi(String garansi) {
      this.garansi = garansi;
   }

   public String getStok() {
      return this.stok;
   }

   public void setStok(String stok) {
      this.stok = stok;
   }

}

File: python/program/Barang.py
#kelas induk

class Barang:
     
    def __init__(self, kode: str, nama: str, harga: str, merk: str, garansi: str, stok: str):
        self._kode = str(kode)
        self.nama = str(nama)
        self.harga = str(harga)
        self.merk = str(merk)
        self._garansi = str(garansi)
        self._stok = str(stok)

    def getKode(self) -> str:
        return self._kode
    
    def setKode(self, kode: str) -> None:
        self._kode = str(kode)

    def getNama(self) -> str:
        return self.nama
    
    def setNama(self, nama: str) -> None:
        self.nama = str(nama)

    def getHarga(self) -> int:
        return self.harga
    
    def setHarga(self, harga: str) -> None:
        self.harga = str(harga)

    def getStok(self) -> str:
       return self._stok

    def setStok(self, stok: str) -> None:
       self._stok = str(stok)

    def getMerk(self) -> str:
       return self.merk

    def setMerk(self, merk: str) -> None:
       self.merk = str(merk)

    def getGaransi(self) -> str:
       return self._garansi

    def setGaransi(self, garansi: str) -> None:
       self._garansi = str(garansi)

